src/handlers.js

```diff
@@ -120,17 +120,18 @@ Handlers.prototype.completed = function(message) {
 
   // Get task definition
   return this.queue.task(message.payload.status.taskId).then(function(task) {
+    
     // Create default expiration date
-    var expires = new Date();
-    expires.setDate(expires.getDate() + 365);
+    var expires = new Date(task.created);
+	expires.setDate(expires.getDate() + 365);
 
```

Comment:
Didn't we miss the whole point of this PR, to use task.expires, if it is available?